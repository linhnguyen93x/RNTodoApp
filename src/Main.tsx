import AppButton from 'components/AppButton'
import * as React from 'react'
import {Text, View} from 'react-native'


interface State {
  message: string
  count: number
}

export default class Main extends React.Component<{}, State> {
  state = {
    message: 'this is a typescript counter',
    count: 0,
  }

  render() {
    return (
      <View style={{
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'space-around',
      }}>
        <Text>{this.state.message}</Text>

        <Text>Count: {this.state.count}</Text>

        <AppButton
          title={'increment'}
          onPress={() => this.setState({count: this.state.count + 1})}
        />
      </View>
    )
  }
}
