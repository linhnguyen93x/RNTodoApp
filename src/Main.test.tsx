import Main from 'Main'
import * as React from 'react'
import 'react-native'
import * as renderer from 'react-test-renderer'

describe('Main snapshot', () => {
  it('renders',  () => {
    const tree = renderer.create(<Main />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
