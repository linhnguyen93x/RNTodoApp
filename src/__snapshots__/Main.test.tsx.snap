// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main snapshot renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "space-around",
    }
  }
>
  <Text>
    this is a typescript counter
  </Text>
  <Text>
    Count: 
    0
  </Text>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        increment (Android)
      </Text>
    </View>
  </View>
</View>
`;
